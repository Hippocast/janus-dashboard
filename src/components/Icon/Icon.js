import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';

import './Icon.css';

export const b = block('j-icon');

const propTypes = {
    ariaLabel: PropTypes.string,
    className: PropTypes.string,
    type: PropTypes.oneOf([
        'add', // green plus
        'checked',
        'close',
        'copy', // green files
        'copy-white', // white files
        'correct',
        'delete', // red bucket
        'delete-white', // white bucket
        'edit', // green pencil
        'github',
        'remove', // red minus
        'successful-white', // white success symbol
    ]).isRequired,
};

const Icon = ({ ariaLabel, className, type }) => (
    <span className={b({ type }).mix(className).mix(ariaLabel && 'j-tooltiped')} aria-label={ariaLabel}/>
);

Icon.propTypes = propTypes;

export default Icon;
