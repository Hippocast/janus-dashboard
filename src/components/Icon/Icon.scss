@import 'index.scss';
$control-size: 20px;
$red-scarlet: #d0011b;

$urls: (
    edit: url(../../assets/img/edit.svg),
    view: url(../../assets/img/eye.svg),
    github: url(../../assets/img/github-icon.svg),
    checked: url(../../assets/img/checked.svg),
    delete: url(../../assets/img/delete.svg),
    delete-white: url(../../assets/img/delete-white.svg),
    close: url(../../assets/img/close.svg),
    copy: url(../../assets/img/copy.svg),
    copy-white: url(../../assets/img/copy-white.svg),
    correct: url(../../assets/img/correct.svg),
    add: url(../../assets/img/plus.svg),
    successful-white: url(../../assets/img/successful-white.svg),
);

.j-icon {
    display: block;
    width: $control-size;
    height: $control-size;
    background-repeat: no-repeat;
    background-position: 50% 50%;
    background-size: 100%;

    &--type-remove {
        height: ceil($control-size/10);
        background-color: $red-scarlet;
    }
}

@each $type, $url in $urls {
    .j-icon--type-#{$type} {
        background-image: $url;
    }
}
