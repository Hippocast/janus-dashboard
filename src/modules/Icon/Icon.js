import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';

import './Icon.css';

const b = block('j-icon');

const propTypes = {
    className: PropTypes.string,
    type: PropTypes.oneOf([
        'add', // green plus
        'checked',
        'close',
        'copy', // green files
        'copy-white', // white files
        'correct',
        'delete', // red bucket
        'delete-white', // white bucket
        'edit', // green pencil
        'github',
        'remove', // red minus
    ]).isRequired,
};

const Control = ({ className, type }) => (
    <span className={b({ type }).mix(className)} />
);

Control.propTypes = propTypes;

export default Control;
