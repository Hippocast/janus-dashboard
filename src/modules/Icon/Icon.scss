$control-size: 20px;
$red-scarlet: #d0011b;

$urls: (
    edit: url(../../assets/img/edit.svg),
    checked: url(../../assets/img/checked.svg),
    delete: url(../../assets/img/delete.svg),
    delete-white: url(../../assets/img/delete-white.svg),
    copy: url(../../assets/img/copy.svg),
    add: url(../../assets/img/plus.svg),
);

.j-icon {
    display: block;
    width: $control-size;
    height: $control-size;
    background-repeat: no-repeat;
    background-position: 50% 50%;

    &--type-remove {
        height: ceil($control-size/10);
        background-color: $red-scarlet;
    }
}

@each $type, $url in $urls {
    .j-icon--type-#{$type} {
        background-image: $url;
    }
}
