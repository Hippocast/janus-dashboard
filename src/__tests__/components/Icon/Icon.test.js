import React from 'react';
import renderer from 'react-test-renderer';

import Icon, { b } from '../../../components/Icon/Icon';

describe('Icon component', () => {
    const render = props => renderer.create(<Icon {...props} />).toJSON();
    const requiredProps = {
        type: 'add',
    };

    it('renders correctly', () => {
        const props = requiredProps;
        const tree = render(props);
        const className = tree.props.className;

        expect(className).toContain('j-icon');
        expect(className).toContain('j-icon--type-add');
        expect(tree).toMatchSnapshot();
    });

    it('renders with an additional className if passed', () => {
        const props = {
            ...requiredProps,
            className: 'mock-class',
        };
        const tree = render(props);
        const className = tree.props.className;

        expect(className).toContain('j-icon');
        expect(className).toContain('j-icon--type-add');
        expect(className).toContain('mock-class');
        expect(tree).toMatchSnapshot();
    });


    it('renders with an additional ariaLabel if passed', () => {
        const props = {
            ...requiredProps,
            ariaLabel: 'mock-label',
        };
        const tree = render(props);
        const className = tree.props.className;

        expect(className).toContain('j-icon');
        expect(className).toContain('j-icon--type-add');
        expect(tree.props['aria-label']).toContain(props.ariaLabel);
        expect(tree).toMatchSnapshot();
    });

    it('renders with an additional className and ariaLabel if passed', () => {
        const props = {
            ...requiredProps,
            className: 'mock-class',
            ariaLabel: 'mock-label',
        };
        const tree = render(props);
        const className = tree.props.className;

        expect(className).toContain('j-icon');
        expect(className).toContain('j-icon--type-add');
        expect(className).toContain('mock-class');
        expect(tree.props['aria-label']).toContain(props.ariaLabel);
        expect(tree).toMatchSnapshot();
    });

});
