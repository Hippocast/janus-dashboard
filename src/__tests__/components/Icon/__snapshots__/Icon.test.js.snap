// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<span
  aria-label={undefined}
  className="j-icon j-icon--type-add"
/>
`;

exports[`Icon component renders with an additional ariaLabel if passed 1`] = `
<span
  aria-label="mock-label"
  className="j-icon j-icon--type-add j-tooltiped"
/>
`;

exports[`Icon component renders with an additional className and ariaLabel if passed 1`] = `
<span
  aria-label="mock-label"
  className="j-icon j-icon--type-add mock-class j-tooltiped"
/>
`;

exports[`Icon component renders with an additional className if passed 1`] = `
<span
  aria-label={undefined}
  className="j-icon j-icon--type-add mock-class"
/>
`;
